packages/extension/src/newtab/DndModal.tsx

```diff
@@ -0,0 +1,129 @@
+import React from 'react';
+import classnames from 'classnames';
+import {
+  getDefaultLink,
+  CustomTime,
+  TimeFormat,
+  getExpiration,
+  getTimeFormatExpiration,
+} from './dnd';
+import { Radio } from '../../../shared/src/components/fields/Radio';
+import { Dropdown } from '../../../shared/src/components/fields/Dropdown';
+import { TextField } from '../../../shared/src/components/fields/TextField';
+import { ModalCloseButton } from '../../../shared/src/components/modals/ModalCloseButton';
+import {
+  StyledModal,
+  ModalProps,
+} from '../../../shared/src/components/modals/StyledModal';
+import styles from './DoNotDisturbModal.module.css';
+import DndContext from './DndContext';
+
+export interface DoNotDisturbModalProps extends ModalProps {
+  defaultTimeFormat?: TimeFormat;
+}
+
+const timeFormatOptions = Object.values(TimeFormat).map((value) => ({
+  label: value,
+  value,
+}));
+
+const customTimeOptions = Object.values(CustomTime);
+
+const DoNotDisturbModal: React.FC<DoNotDisturbModalProps> = ({
+  defaultTimeFormat = TimeFormat.HALF_HOUR,
+  onRequestClose,
+  ...modalProps
+}) => {
+  const { setDndSettings, isActive } = React.useContext(DndContext);
```

Comment:
We usually use directly the `useContext`, `useState`, and all the others hooks without using the React namespace.